deployment/src/main/java/io/quarkiverse/cxf/deployment/QuarkusCxfProcessor.java

```diff
@@ -212,93 +222,89 @@ public void build(
                 continue;
             }
 
             ClassInfo wsClassInfo = annotation.target().asClass();
+
             String sei = wsClassInfo.name().toString();
+
             reflectiveClass
                     .produce(new ReflectiveClassBuildItem(true, true, sei));
             unremovableBeans.produce(new UnremovableBeanBuildItem(
                     new UnremovableBeanBuildItem.BeanClassNameExclusion(sei)));
-            List<String> wrapperClassNames = new ArrayList<>();
 
             if (!Modifier.isInterface(wsClassInfo.flags())) {
                 continue;
             }
-            QuarkusJaxWsServiceFactoryBean jaxwsFac = new QuarkusJaxWsServiceFactoryBean();
-            Bus bus = BusFactory.getDefaultBus();
-            jaxwsFac.setBus(bus);
-            bus.setExtension(c, GeneratedClassClassLoaderCapture.class);
-            //TODO here add all class
-            try {
-                jaxwsFac.setServiceClass(Thread.currentThread().getContextClassLoader().loadClass(sei));
-                jaxwsFac.create();
-                wrapperClassNames.addAll(jaxwsFac.getWrappersClassNames());
-            } catch (ClassNotFoundException e) {
-                LOGGER.error("failed to load WS class : " + sei);
-            }
-            String pkg = wsClassInfo.name().toString();
-            int idx = pkg.lastIndexOf('.');
-            if (idx != -1 && idx < pkg.length() - 1) {
-                pkg = pkg.substring(0, idx);
-            }
-            AnnotationValue namespaceVal = annotation.value("targetNamespace");
-            String wsNamespace = namespaceVal != null ? namespaceVal.asString() : getNamespaceFromPackage(pkg);
-            String wsName = "";
-            if (annotation.value("serviceName") != null) {
-                wsName = annotation.value("serviceName").asString();
-            }
-            Collection<ClassInfo> implementors = index.getAllKnownImplementors(DotName.createSimple(sei));
+
+            // created on-demand if an implementor or client usage is found
+            QuarkusJaxWsServiceFactoryBean factoryBean = null;
+
+            String wsNamespace = Optional.ofNullable(annotation.value("targetNamespace"))
+                    .map(AnnotationValue::asString)
+                    .orElseGet(() -> getNameSpaceFromClassInfo(wsClassInfo));
 
             //TODO add soap1.2 in config file
-            String soapBinding = SOAPBinding.SOAP11HTTP_BINDING;
-            //if no implementor, it mean it is client
-            if (implementors == null || implementors.isEmpty()) {
+            final String soapBindingDefault = SOAPBinding.SOAP11HTTP_BINDING;
+
+            Collection<ClassInfo> implementors = index.getAllKnownImplementors(DotName.createSimple(sei));
+
+            if (implementors != null && !implementors.isEmpty()) {
+                factoryBean = factoryBean == null ? createQuarkusJaxWsServiceFactoryBean(sei, bus) : factoryBean;
```

Comment:
The creation of the `factoryBean` is lazy to make sure it is only created when _really required_, that is when an implementor and/or a client is found.

Reply:
See also these two lines further up: https://github.com/quarkiverse/quarkus-cxf/pull/262/files#diff-ab6082b1f5a32014ec91e75ab14715d38b1d3e2bd2cab287562542c1af6a37b6R258-R259

Reply:
What I meant is that factoryBean will always be null on line 271 (so it's a redundant check) - and I believe for line 286 possibly as well?

Reply:
No, because there is a loop over (possibly) multiple implementors and the factoryBean shall be:
- created laziliy
- and only once

So I have to cover:
- neither impl nor client -> don't create bean at all
- create once on first impl and reuse for subsequent impls (and client, if required)
- create for client in case no impl is found

Reply:
OK, I may be seeing something wrong then, my apologies if so ... what I'm seeing is that you are setting the factoryBean to null (line 259) **inside the for loop, and for every iteration** - so I don't know how you are for example, creating it on first impl and reusing for subsequent impls ...

Reply:
Watch out, there are two nested loops! (which might be another candidate for refactoring)

You have an outer loop for each SEI and per SEI there is also a loop over all impls of that SEI.

Reply:
I think as long as you're sure of the flow here, we can resolve this ... I just wasn't seeing the case where factoryBean would be something other than null!  But you've verified, so I think we're good!